Source/WebCore/platform/audio/SharedAudioDestination.cpp

```diff
@@ -0,0 +1,315 @@
+    // renderers is empty, do not wait for the render thread to
+    // finish configuration, as it will never run.
+    bool shouldSkipRendering = !m_started && m_renderers.isEmpty();
+
+    if (!shouldSkipRendering) {
+        // The AudioDestination must be started for configuration to take place.
+        if (shouldStart) {
+            // Dispatching to the main thread is required for destinations
+            // which are subclasses of AudioDestinationResampler.
+            callOnMainThreadAndWait([this] {
+                m_destination->start(nullptr, [] (bool) { });
+            });
+            m_started = true;
+        }
+
+        m_configurationSemaphore.wait();
+
+        // The AudioDestination must not be stopped before configuration takes place.
+        if (shouldStop) {
+            // Dispatching to the main thread is required for destinations
+            // which are subclasses of AudioDestinationResampler.
+            callOnMainThreadAndWait([this] {
+                m_destination->stop();
+            });
+            m_started = false;
+        }
+    }
+
+    // Move the previously configured list of renderers to the MainThread for destruction:
+    Vector<RefPtr<SharedAudioDestination>> renderersToDispose;
+    {
+        LockHolder locker { m_renderLock };
+        renderersToDispose = std::exchange(m_newRenderers, { });
+    }
+
+    callOnMainThread([completionHandler = WTFMove(completionHandler), renderersToDispose = WTFMove(renderersToDispose)]() mutable {
+        renderersToDispose.clear();
+        completionHandler(true);
+    });
+}
+
+void SharedAudioDestinationAdapter::render(AudioBus* sourceBus, AudioBus* destinationBus, size_t numberOfFrames, const AudioIOPosition& outputPosition)
+{
+    if (m_renderLock.tryLock()) {
+        Locker locker { AdoptLock, m_renderLock };
+        if (m_needsConfiguration) {
+            // The SharedAudioDestinationAdapter avoids allocing or deallocing on the
+            // high priority audio thread by merely swapping the contents of the renderer
+            // configuration vectors. After the swap, the contents of m_newRenderers will
+            // be destroyed by configureRenderThread() on the m_configurationWorkQueue.
+            std::swap(m_newRenderers, m_configuredRenderers);
+            m_needsConfiguration = false;
+            m_configurationSemaphore.signal();
+        }
+    }
+
+    bool firstRenderer = true;
+    for (auto& renderer : m_configuredRenderers) {
+        if (firstRenderer) {
+            // The first renderer should render directly to destinationBus.
+            renderer->sharedRender(sourceBus, destinationBus, numberOfFrames, outputPosition);
+            firstRenderer = false;
+            continue;
+        }
+        // Subsequent renderers should render to the m_workBus, which will
+        // then be summed to the destinationBus.
+        renderer->sharedRender(sourceBus, m_workBus.ptr(), numberOfFrames, outputPosition);
```

Comment:
You need a Ref to the workBus on the stack before passing it to the function, as per our smart pointer guidelines. Likely want to add a new `protectedWorkBus()` getter which returns a `Ref` and call `protectedWorkBus().ptr()` here.